src/psyclone/psyir/backend/visitor.py

```diff
@@ -152,7 +152,35 @@ def __call__(self, node):
         :rtype: str
 
         '''
-        return self._visit(node)
+        if not isinstance(node, Node):
+            raise TypeError(
+                "The PSyIR visitor functor method only accepts a PSyIR Node "
+                "as argument, but found '{0}'.".format(type(node).__name__))
+
+        # The visitor must not have any side effect to the provided node, but
```

Comment:
"...must not alter the provided node but if there are any DSL concepts then these will..."

Reply:
Done